src/electron/routing_service.ts

```diff
@@ -200,18 +210,21 @@ export class RoutingDaemon {
   // stop() resolves when the stop command has been sent.
   // Use #onceDisconnected to be notified when the connection terminates.
   async stop() {
-    if (!this.socket) {
-      // Never started.
+    try {
+      if (!this.socket) {
+        // Never started.
+        return;
+      }
+      if (this.stopping) {
+        // Already stopped.
+        return;
+      }
+      this.stopping = true;
+
+      return this.writeReset();
+    } finally {
       this.fulfillDisconnect();
```

Comment:
This might not be accurate because `fulfillDisconnect()` shall be called in `cleanup`.